spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/watcher/DependencyWatcher.java

```diff
@@ -0,0 +1,34 @@
+package org.springframework.cloud.zookeeper.discovery.watcher;
+
+import java.io.IOException;
+
+public interface DependencyWatcher {
```

Comment:
The `DependencyWatcher` interface is a bit confusing to me. It appears to mix lifecycle methods (`registerDependencyRegistrationHooks`, `clearDependencyRegistrationHooks`) with listener registration methods. Would an end user ever invoke these methods directly, or would it all be invoked by the thread that initializes the application context?